python-backend/app/api/mines/tailings/resources/tailings.py

```diff
@@ -45,48 +45,48 @@ def get(self, mine_tailings_storage_facility_guid=None):
     def post(self, mine_tailings_storage_facility_guid=None):
         if not mine_tailings_storage_facility_guid:
             data = self.parser.parse_args()
-            mine_guid = data['mine_guid']
+            mine_guid = data.get('mine_guid')
             # see if this would be the first TSF
             mine_tsf_list = MineTailingsStorageFacility.find_by_mine_guid(mine_guid)
             is_mine_first_tsf = len(mine_tsf_list) == 0
             mine_tsf = MineTailingsStorageFacility(
                 mine_guid=mine_guid,
-                mine_tailings_storage_facility_name=data['tsf_name'],
+                mine_tailings_storage_facility_name=data.get('tsf_name'),
                 **self.get_create_update_dict())
             db.session.add(mine_tsf)
             if is_mine_first_tsf:
                 try:
                     req_documents_url = get_documents_svc_url('/required?category=MINE_TAILINGS')
                     get_tsf_docs_resp = requests.get(
                         req_documents_url,
-                        headers={'Authorization': request.headers['Authorization']})
+                        headers={'Authorization': request.headers.get('Authorization')})
 
                     if get_tsf_docs_resp.status_code != 200:
                         self.raise_error(
                             500,
                             'get_tsf_req_docs returned error' + str(get_tsf_docs_resp.status_code))
 
-                    tsf_required_documents = get_tsf_docs_resp.json()['required_documents']
+                    tsf_required_documents = get_tsf_docs_resp.json().get('required_documents')
```

Comment:
for example, now that tsf_required_documents can be none, the loop below will silently be skipped, rather than crashing should something be changed on other endpoint.